packages/network-controller/src/provider-api-tests/helpers.ts

```diff
@@ -0,0 +1,479 @@
+}
+
+type MockBlockTrackerRequestOptions = {
+  /**
+   * A nock scope (a set of mocked requests scoped to a certain base url).
+   */
+  nockScope: NockScope;
+  /**
+   * The block number that the block tracker should report, as a 0x-prefixed hex string.
+   */
+  blockNumber: string;
+
+  block: any;
+};
+
+const mockNextBlockTrackerRequest = ({
+  nockScope,
+  blockNumber = DEFAULT_LATEST_BLOCK_NUMBER,
+  block = DEFAULT_BLOCK,
+}: MockBlockTrackerRequestOptions) => {
+  // eslint-disable-next-line
+  mockRpcCall({
+    nockScope,
+    request: { method: 'eth_blockNumber', params: [] },
+    response: { result: blockNumber },
+  });
+
+  // eslint-disable-next-line
+  mockRpcCall({
+    nockScope,
+    request: { method: 'eth_getBlockByNumber', params: [blockNumber, false] },
+    response: {
+      result: { ...block, number: blockNumber, hash: `0x${Math.random()}` },
+    },
+    // times: 2,
+  }) as NockScope;
+};
+
+const mockAllBlockTrackerRequests = ({
+  nockScope,
+  blockNumber = DEFAULT_LATEST_BLOCK_NUMBER,
+  block = DEFAULT_BLOCK,
+}: MockBlockTrackerRequestOptions) => {
+  (
+    mockRpcCall({ // eslint-disable-line
+      nockScope,
+      request: { method: 'eth_blockNumber', params: [] },
+      response: { result: blockNumber },
+    }) as NockScope
+  ).persist();
+
+  (
+    mockRpcCall({ // eslint-disable-line
+      nockScope,
+      request: { method: 'eth_getBlockByNumber', params: [blockNumber, false] },
+      response: {
+        result: { ...block, number: blockNumber, hash: `0x${Math.random()}` },
+      },
+    }) as NockScope
+  ).persist();
+};
+
+type Request = { method: string; params?: any[] };
+type Response = {
+  error?: any;
+  result?: any;
+  httpStatus?: number;
+  [k: string]: any;
+};
+type ResponseBody = { body: JSONRPCResponse };
+type BodyOrResponse = ResponseBody | Response;
+type CurriedMockRpcCallOptions = {
+  request: Request;
+  // The response data.
+  response?: BodyOrResponse;
+  /**
+   * An error to throw while making the request.
+   * Takes precedence over `response`.
+   */
+  error?: Error | string;
+  /**
+   * The amount of time that should pass before the
+   * request resolves with the response.
+   */
+  delay?: number;
+  /**
+   * The number of times that the request is
+   * expected to be made.
+   */
+  times?: number;
+};
+
+type MockRpcCallOptions = {
+  // A nock scope (a set of mocked requests scoped to a certain base URL).
+  nockScope: nock.Scope;
+} & CurriedMockRpcCallOptions;
+
+type MockRpcCallResult = nock.Interceptor | nock.Scope;
+
+const mockRpcCall = ({
+  nockScope,
+  request,
+  response,
+  error,
+  delay,
+  times,
+}: MockRpcCallOptions): MockRpcCallResult => {
+  // eth-query always passes `params`, so even if we don't supply this property,
+  // for consistency with makeRpcCall, assume that the `body` contains it
+  const { method, params = [], ...rest } = request;
+  const httpStatus = (response as Response)?.httpStatus ?? 200;
```

Comment:
Is the typecast necessary here?

Reply:
well its not necessary, but the alternatives are quite terse for what it is

Reply:
nvm fixed it.